node_modules/oae-tenants/lib/api.js

```diff
@@ -142,10 +159,71 @@ var getTenants = module.exports.getTenants = function(excludeDisabled) {
             filteredTenants[tenantAlias] = _copyTenant(tenant);
         }
     });
+
     return filteredTenants;
 };
 
 /**
+ * Search for tenants based on a full-text search query
+ *
+ * @param  {String}         [q]                 The full-text query to perform. If unspecified, all tenants will be returned
+ * @param  {Object}         [opts]              Optional arguments
+ * @param  {Number}         [opts.start]        The index at which to begin returning results
+ * @param  {Number}         [opts.limit]        The maximum number of results to return. If unspecified, returns all results
+ * @param  {Boolean}        [opts.disabled]     If `true`, will include tenants that are disabled or deleted. Otherwise, they are ommitted from results
+ * @return {SearchResult}                       The search result object containing the tenants
+ */
+var searchTenants = module.exports.searchTenants = function(q, opts) {
+    q = (_.isString(q)) ? q.trim() : null;
+    opts = opts || {};
+    opts.start = OaeUtil.getNumberParam(opts.start, 0);
+
+    // Determine if we should included disabled/deleted tenants
+    var includeDisabled = (_.isBoolean(opts.disabled)) ? opts.disabled : false;
+
+    // Create a sorted result of tenants based on the user's query. If there was no query, we will
+    // pull the pre-sorted list of tenants from the global cache
+    var results = null;
+    if (q) {
+        results = _.chain(tenantSearchIndex.search(q))
+            .sortBy('ref')
+            .sortBy('score')
+            .pluck('ref')
+            .map(getTenant)
+            .value();
+    } else {
+        results = sortedTenants;
+    }
+
+    // Filter out the global admin server and deleted tenancies
+    results = _.filter(results, function(tenant) {
+        return (!tenant.isGlobalAdminServer);
+    });
+
+    // Filter out disabled or deleted tenants if they aren't said to be included
+    if (!includeDisabled) {
+        results = _.where(results, {'active': true, 'deleted': false});
+    }
```

Comment:
I think I had only done this to handle `isGlobalAdminServer` safer with a `false`y check instead of strict `false` check. There shouldn't be any issue implementing this suggestion.